src/hb-aat-layout-trak-table.hh

```diff
@@ -82,9 +120,38 @@ struct trak
   inline bool sanitize (hb_sanitize_context_t *c) const
   {
     TRACE_SANITIZE (this);
+    /* TODO */
     return_trace (c->check_struct (this));
   }
 
+  inline bool apply (hb_aat_apply_context_t *c) const
+  {
+    TRACE_APPLY (this);
+    const float ptem = c->font->ptem;
+    if (ptem > 0.f)
+    {
+      hb_buffer_t *buffer = c->buffer;
+      if (HB_DIRECTION_IS_HORIZONTAL (c->buffer->props.direction))
+      {
+        const TrackData trackData = this+horizOffset;
+        float tracking = trackData.get_tracking (this, ptem);
+        int x_scale = c->font->x_scale / c->face->upem;
+        // TODO: Do we anything else for RTL?
```

Comment:
https://github.com/qdot/gecko-hg/blob/ab17432d9bede83cf25be231f4bfba44478561fb/gfx/thebes/gfxFont.cpp#L743 for example

Reply:
Ditto below